Framework/PythonInterface/plugins/algorithms/LoadElementalAnalysisData.py

```diff
@@ -0,0 +1,179 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+
+import glob
+import os
+import numpy as np
+from mantid import config
+from mantid.api import AlgorithmFactory, AnalysisDataService, PythonAlgorithm, TextAxis, WorkspaceGroup, WorkspaceGroupProperty
+from mantid.kernel import Direction, IntBoundedValidator
+from mantid.simpleapi import ConvertToHistogram, ConvertToPointData, DeleteWorkspace, LoadAscii, WorkspaceFactory
+
+
+type_keys = {"10": "Prompt", "20": "Delayed", "99": "Total"}
+spectrum_index = {"Delayed": 1, "Prompt": 2, "Total": 3}
+num_files_per_detector = 3
+run_number_length = 5
+
+
+class LoadElementalAnalysisData(PythonAlgorithm):
+    def category(self):
+        """
+        Returns category
+        """
+        return "DataHandling"
+
+    def PyInit(self):
+        self.declareProperty(name='Run', defaultValue=0, validator=IntBoundedValidator(lower=0),
+                             doc='Run number to load')
+
+        self.declareProperty(WorkspaceGroupProperty(name='GroupWorkspace', defaultValue='',
+                                                    direction=Direction.Output),
+                             doc='Output group workspace for run')
+
+    def validateInputs(self):
+        issues = dict()
+        checkRun = self.search_user_dirs()
+        if not checkRun:
+            issues['Run'] = "Cannot find files for run " + self.getPropertyValue("Run")
+        return issues
+
+    def PyExec(self):
+        run = self.getPropertyValue("Run")
+        to_load = self.search_user_dirs()
+        workspaces = {
+            path: self.get_filename(path, run)
+            for path in to_load if self.get_filename(path, run) is not None
+        }
+        unique_workspaces = {}
+        for path, workspace in workspaces.items():
+            if workspace not in unique_workspaces.values():
+                unique_workspaces[path] = workspace
+        workspaces = unique_workspaces
+
+        self.format_workspace(workspaces)
+        self.merge_workspaces(workspaces.values())
+
+    def pad_run(self):
+        """ Pads run number: i.e. 123 -> 00123; 2695 -> 02695 """
+        return str(self.getProperty("Run").value).zfill(run_number_length)
+
+    def search_user_dirs(self):
+        """ Finds all files for the run number provided """
+        files = []
+        for user_dir in config["datasearch.directories"].split(";"):
```

Comment:
Can I suggest we use [config.getDataSearchDirs()](https://github.com/mantidproject/mantid/blob/55f8921ce601f94a99678ccaaf0e760cd5b8e455/Framework/PythonInterface/mantid/kernel/src/Exports/ConfigService.cpp) instead of splitting the string ourselves? It hides the detail that `;` is a separator.

Reply:
Resolved in commit fd81a40